src/libsyntax/feature_gate.rs

```diff
@@ -504,6 +504,10 @@ declare_features! (
 
     // `reason = ` in lint attributes and `expect` lint attribute
     (active, lint_reasons, "1.31.0", Some(54503), None),
+
+    // Allows calling `const unsafe fn` and constructing layout-restricted types like `NonZero`
```

Comment:
This comment does not match reality: You currently allow constructing layout-restricted types even without the feature flag.

I'd prefer if we did what the comment said, but at least they should be consistent.

Reply:
I don't think we can do what the comment says due to stability, sadly ;)

Reply:
I changed the comment

Reply:
> I don't think we can do what the comment says due to stability, sadly ;)

Why not? Just enable this `min_unsafe_const_fn` feature in libstd?

Reply:
I think @Centril is referring to the fact that you can't use unstable const features in `min_const_fn` functions. But these unsafety checks are an exception.

That said, I think there's little point to this, as these are technically already behind the `rustc_attrs` feature gate, as you can't define such layout restricted types without unstable attributes. They can only really be used in the compiler or other unstable code. Having to enable another feature gate doesn't protect anyone from using unstable features.